app/uk/gov/hmrc/excisemovementcontrolsystemapi/connectors/EISSubmissionConnector.scala

```diff
@@ -49,7 +50,8 @@ class EISSubmissionConnector @Inject()
     //todo: add retry
     val correlationId = emcsUtils.generateCorrelationId
     val createdDateTime = emcsUtils.getCurrentDateTimeString
-    val encodedMessage = emcsUtils.createEncoder.encodeToString(request.body.toString.getBytes(StandardCharsets.UTF_8))
+    val wrappedXml: NodeSeq = wrapXmlInControlDocument(request)
+    val encodedMessage = emcsUtils.createEncoder.encodeToString(wrappedXml.toString.getBytes(StandardCharsets.UTF_8))
```

Comment:
There is a function in emcsUtils that does the encoding of a string. Can we use that function here, please. I thing this was missed when we created that function in emcsUtils.

Reply:
Used the function in EMCS Utils. The EMCS Utils class doesn't seem like a very Scala way of doing things. It might have been better to define traits for each of the "utility" methods then we could have provided test versions of those traits. We could have an encode trait, a decode trait, a timestamp provider trait etc. rather than one "static" Java style class??

Reply:
I do not always agree with the Scala way (create treat, using inheritance for sharing method of info), as this can have his inconveniences as a class can end up deriving for multiple trait which became more difficult  to test, and maybe import stuff that it does not need. I personally prefer composition over inheritance. I see in the past classes deriving from multiple treat and the same trait deriving from the same trait, which complicate stuff. My point was to keep it DRY and reused the decode or encode function. Yes I think in this case the encode and decode method of the emcsClass may be put in a trait as this class is becoming big and method do not related,  but I think the new decode/encode trait should only define these two method.